### PR TITLE
feat(docs): rework roadmap based on design

### PR DESCRIPTION
**Pull request by CollierCZ**
In the masonry layout, items get added to columns based on their order, rather than height, so the result isn't always perfect. But checking based on height is slow, so it seemed acceptable. Let me know if you'd prefer the other way.
 Orbit.kiwi: https://orbit-docs-docs-rework-roadmap.surge.sh
 Storybook: https://orbit-docs-rework-roadmap.surge.sh

### CODE REVIEW

File: docs/src/pages/roadmap.tsx
Comment (silvenon) [inline docs/src/pages/roadmap.tsx:24]: Use smart apostrophes, like above:
```suggestion
      what’s already designed and the state it’s in, visit{" "}
```
There's one more below.

**Review by silvenon — CHANGES_REQUESTED** [review]
I'm wondering what to do about the TOC wrapper when there's no TOC 🤔  I think it's alright to still reserve that space to keep design consistency, but there shouldn't be an empty "Table of contents" title on mobile.

File: docs/src/components/MasonryLayout.tsx
Comment (silvenon) [reply docs/src/components/MasonryLayout.tsx:50]: Just to add a better reason: using inline styles won't include vendor prefixes for declarations that need them, like Flexbox.

Comment (CollierCZ) [reply docs/src/components/MasonryLayout.tsx:50]: Yes, I'd planned to get rid of this, just overlooked it once I got the feature working. I'll keep your love of Tailwind just between us.

File: docs/src/components/MasonryLayout.tsx
Comment (CollierCZ) [reply docs/src/components/MasonryLayout.tsx:26]: Yes! Meant to do it, but slipped my mind.

**Comment by CollierCZ** [comment]
> I'm wondering what to do about the TOC wrapper when there's no TOC 🤔 I think it's alright to still reserve that space to keep design consistency, but there shouldn't be an empty "Table of contents" title on mobile.

Not sure about the space on the right, but I can definitely remove the collapse on mobile.

**Comment by CollierCZ** [comment]
I realized I already had a check to hide the ToC when it wasn't populated, but then its structured change. So I fixed the check. And it seemed to hide the space on the right too, so that's how it is for now.

**Review by silvenon — DISMISSED** [review]
After this you can merge right away.

**Pull request merged**
